core-java-modules/core-java-concurrency-advanced-4/src/main/java/com/baeldung/producerconsumer/DataQueue.java

```diff
@@ -47,13 +51,20 @@ public void notifyAllForEmpty() {
 
     public void add(Message message) {
         synchronized (queue) {
+            if(queue.size() > maxSize) {
+                System.out.println("Max size exceeded");
+            }
             queue.add(message);
         }
     }
 
     public Message remove() {
         synchronized (queue) {
-            return queue.poll();
+            final Message poll = queue.poll();
```

Comment:
Rollback any changes used for testing requester's claim

Reply:
Done